From 1fdd8c04b65bd55730e6931a520cc1dabfc4d190 Mon Sep 17 00:00:00 2001
From: John Crispin <blogic@openwrt.org>
Date: Thu, 16 Feb 2012 20:17:16 +0100
Subject: [PATCH 31/70] SERIAL: MIPS: lantiq: convert serial driver to clkdev
 api

Reference the FPI clock via its new access function.

Signed-off-by: John Crispin <blogic@openwrt.org>
Cc: linux-serial@vger.kernel.org
---
 drivers/tty/serial/lantiq.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

--- a/drivers/tty/serial/lantiq.c
+++ b/drivers/tty/serial/lantiq.c
@@ -540,6 +540,10 @@ lqasc_request_port(struct uart_port *por
 		if (ltq_gpio_request(&pdev->dev, MUXC_SIF_TX_PIN,
 				3, 1, "asc1-tx"))
 			return -EBUSY;
+		ltq_port->clk = clk_get(&pdev->dev, NULL);
+		if (IS_ERR(ltq_port->clk))
+			return PTR_ERR(ltq_port->clk);
+		clk_enable(ltq_port->clk);
 	}
 	return 0;
 }
@@ -698,7 +702,7 @@ lqasc_probe(struct platform_device *pdev
 	if (lqasc_port[pdev->id] != NULL)
 		return -EBUSY;
 
-	clk = clk_get(&pdev->dev, "fpi");
+	clk = clk_get_fpi();
 	if (IS_ERR(clk)) {
 		pr_err("failed to get fpi clk\n");
 		return -ENOENT;
